### ISSUE TITLE
Plugin: Synchronize PHP changes for WordPress 6.5 release

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#57959]
The issue tracks changes to PHP files that (might) need to be backported and merged to Core for the WordPress 6.5 release. The list is exhaustive so it may cover multiple PRs and experimental changes that won't be in 6.5.

## Beta 1 is February 13, 2024

- [Release schedule](https://make.wordpress.org/core/6-5/)
- [6.5 Roadmap](https://make.wordpress.org/core/2023/12/07/roadmap-to-6-5/)

## Tips 👀 

- Check whether [files in your PR really do need backporting](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/back-merging-to-wp-core.md) - if they don't please leave a comment to that effect and apply strikethrough to the line item in question.
- **Please [open a new Core Trac ticket for each PR](https://core.trac.wordpress.org/newticket)** and link it next to the citation in this issue. 🙇 
- For ease of cross-linking and tracking, you can submit your PRs to [WordPress/wordpress-develop](https://github.com/WordPress/wordpress-develop). Refer to docs on [how to fork the repository and create a PR](https://make.wordpress.org/core/handbook/tutorials/trac/submitting-a-patch/#adding-your-github-fork-to-your-wp-trunk-copy). Link any backport PR next to the citation in this issue. 🙇 
- Try to limit PRs to one feature to make reviewing easier and the commit history clear. See [why](https://github.com/WordPress/gutenberg/issues/47187#issuecomment-1384107056)
- ~Modifications/additions to PHP unit tests aren't listed: any unit tests introduced in the cited PRs should also be ported over where applicable.~ These are now listed 🎉 
- Once your backport PR is up, add it to the [WordPress 6.5 Editor Tasks project board](https://github.com/orgs/WordPress/projects/148) so folks can review it. Thank you!
- PRs marked with ⚠️ may have previously been backported to WP Core. Please check these _particularly carefully_. 

❗ _This is a living document and will be constantly updated as we approach beta._ Sorry if you were mistakenly pinged! If a file or files have already been backported, please tick them off and link to the backport PR/trac issue.

Or, if the changes are not slated for WordPress 6.5, you can ~~strike~~ them or make a comment in this issue. Thank you! 🙇

----

PRs to be backported will have the **"Needs PHP backport"** label applied. 
<img width="142" alt="Screenshot 2023-06-05 at 10 47 22 am" src="https://github.com/WordPress/gutenberg/assets/6458278/299596a3-c7c3-4894-8fa9-012156985bda">

# 📁 lib

### 📄 shadow.php

-   [x] https://github.com/WordPress/gutenberg/pull/57654 - @vcanales | [Trac ticket](https://core.trac.wordpress.org/ticket/60377) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5972)

### 📄 pattern.php
(this file has now been removed in trunk)

- [x] https://github.com/WordPress/gutenberg/pull/57912 - @talldan | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/57789 - @kevin940726 | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/57249 - @SantosGuillamot | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/56235 - @kevin940726 | Trac ticket | Core backport PR 

### 📄 typography.php

- [x] https://github.com/WordPress/gutenberg/pull/57866 - @ramonjd | [Trac ticket](https://core.trac.wordpress.org/ticket/60263) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5875) 
- [x] https://github.com/WordPress/gutenberg/pull/54835 - @spacedmonkey | [Trac ticket](https://core.trac.wordpress.org/ticket/59453) | Core backport PR (⚠️ Check for existing WP Core backport)
### 📄 layout.php

- [x] https://github.com/WordPress/gutenberg/pull/58251 - @getdave | [Trac ticket](https://core.trac.wordpress.org/ticket/60327) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5935)
- [x] ~https://github.com/WordPress/gutenberg/pull/57564 - @tellthemachines | Trac ticket | Core backport PR~ 
- [x] https://github.com/WordPress/gutenberg/pull/56171 - @nk-o | [Trac ticket](https://core.trac.wordpress.org/ticket/60327#ticket) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5935) 
- [x] https://github.com/WordPress/gutenberg/pull/56187 - @andrewserong | [Trac ticket](https://core.trac.wordpress.org/ticket/60292) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5900) 
- [x] https://github.com/WordPress/gutenberg/pull/56130 - @tellthemachines | [Trac ticket](https://core.trac.wordpress.org/ticket/60130) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5812) 
- [x] ~https://github.com/WordPress/gutenberg/pull/55416 - @luisherranz | Trac ticket | Core backport PR (Already backported in WP 6.4)~
### 📄 background.php

- [x] https://github.com/WordPress/gutenberg/pull/57495 - @andrewserong | [Trac ticket](https://core.trac.wordpress.org/ticket/60175) |
- [x] https://github.com/WordPress/gutenberg/pull/57005 - @andrewserong | [Trac ticket](https://core.trac.wordpress.org/ticket/60175) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5828)
- [x] https://github.com/WordPress/gutenberg/pull/56997 - @andrewserong | (doesn't need backporting)
- [x] ~https://github.com/WordPress/gutenberg/pull/54850 - @andrewserong | (already in core)~
- [x] ~https://github.com/WordPress/gutenberg/pull/54489 - @andrewserong | (already in core)~

### 📄 behaviors.php

- [x] ~https://github.com/WordPress/gutenberg/pull/57165 - @michalczaplinski | Trac ticket | Core backport PR~ 
- [x] ~https://github.com/WordPress/gutenberg/pull/55385 - @afercia | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~
- [x] ~https://github.com/WordPress/gutenberg/pull/55010 - @afercia | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~
- [x] ~https://github.com/WordPress/gutenberg/pull/53026 - @c4rl0sbr4v0 | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~
### 📄 elements.php

- [x] ~https://github.com/WordPress/gutenberg/pull/55228 - @dmsnell | [Trac ticket](https://core.trac.wordpress.org/ticket/59544) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5411) (⚠️ Check for existing WP Core backport)~

### 📄 dimensions.php

- [x] https://github.com/WordPress/gutenberg/pull/56897 - @andrewserong | [Trac ticket](https://core.trac.wordpress.org/ticket/60365) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5963)

---
### 📁 wordpress-6.5
- [x] https://github.com/WordPress/gutenberg/pull/58488 - @afercia | [Trac ticket](https://core.trac.wordpress.org/ticket/60412#ticket) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/6002)
-   [x] https://github.com/WordPress/gutenberg/pull/56682 - @afercia | [Trac ticket](https://core.trac.wordpress.org/ticket/60105) | ~~Core backport~~ PR (N/A - diff attached to trac ticket)
-   [x] https://github.com/WordPress/gutenberg/pull/57924 - @DAreRodz | Trac ticket | Core backport PR
-   [x] https://github.com/WordPress/gutenberg/pull/58066 - @luisherranz | [Trac ticket](https://core.trac.wordpress.org/ticket/60356#ticket) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5953)
-   [x] ~https://github.com/WordPress/gutenberg/pull/58205 - @michalczaplinski | [Trac ticket](https://core.trac.wordpress.org/ticket/60282) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5888)~
-   [x] ~https://github.com/WordPress/gutenberg/pull/58089 - @michalczaplinski | [Trac ticket](https://core.trac.wordpress.org/ticket/60282) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5888)~
-   [x] ~https://github.com/WordPress/gutenberg/pull/58126 - @getdave | Trac ticket | Core backport PR~
-   [x] ~https://github.com/WordPress/gutenberg/pull/57979 - @scruffian | Trac ticket | Core backport PR~
-   [x] https://github.com/WordPress/gutenberg/pull/57778 - @c4rl0sbr4v0 | Trac ticket | Core backport PR
-   [x] ~https://github.com/WordPress/gutenberg/pull/58112 - @youknowriad | Trac ticket | Core backport PR~
- [x] ~https://github.com/WordPress/gutenberg/pull/57022 - @dmsnell | Trac ticket | Core backport PR~ 
- [x] https://github.com/WordPress/gutenberg/pull/57030 - @anton-vlasenko | [Trac ticket](https://core.trac.wordpress.org/ticket/60078) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5768) 
- [x] ~https://github.com/WordPress/gutenberg/pull/56928 - @dmsnell | Trac ticket | Core backport PR~
- [x] ~https://github.com/WordPress/gutenberg/pull/56952 - @kt-12 | Trac ticket | Core backport PR~
- [x] ~https://github.com/WordPress/gutenberg/pull/57520 - @scruffian | Trac ticket | Core backport PR~ 
- [x] ~https://github.com/WordPress/gutenberg/pull/57362 - @t-hamano | Trac ticket | Core backport PR~ 
- [x] ~https://github.com/WordPress/gutenberg/pull/56629 - @talldan | Trac ticket | Core backport PR~ 
- [x] https://github.com/WordPress/gutenberg/pull/56143 - @luisherranz | [Trac ticket](https://core.trac.wordpress.org/ticket/56313) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5818) 
- [x] https://github.com/WordPress/gutenberg/pull/55459 - @DAreRodz | Trac ticket | Core backport PR 
- [x] ~https://github.com/WordPress/gutenberg/pull/56459 - ~@tellthemachines~ @scruffian  | Trac ticket | Core backport PR~
- [x] ~https://github.com/WordPress/gutenberg/pull/55605 - @scruffian | Trac ticket | Core backport PR~
- [x] [Commit](https://github.com/WordPress/gutenberg/commit/a9f307bf4f9a111b82b31062a4c3a8c34a1b881a) - @scruffian | Trac ticket | Core backport PR 
- [x] [Commit](https://github.com/WordPress/gutenberg/commit/7ce6b179ab049be8b1e5648e5cd2a6eb65d54736) - @scruffian | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/57005 - @andrewserong | [Trac ticket](https://core.trac.wordpress.org/ticket/60175) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5828)
- [x] https://github.com/WordPress/gutenberg/pull/57212 - @t-hamano | [Trac ticket](https://core.trac.wordpress.org/ticket/60119) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5805)
- [x] https://github.com/WordPress/gutenberg/pull/57094 - @youknowriad | [Trac ticket](https://core.trac.wordpress.org/ticket/60322#ticket) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5929) 
- [x] https://github.com/WordPress/gutenberg/pull/55236 - @aaronrobertshaw | [Trac ticket](https://core.trac.wordpress.org/ticket/60342) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5943)
- [x] https://github.com/WordPress/gutenberg/pull/56990 - @jorgefilipecosta | Trac ticket | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5940) 
- [x] https://github.com/WordPress/gutenberg/pull/56567 - @jorgefilipecosta | Trac ticket | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5940) 
- [x] https://github.com/WordPress/gutenberg/pull/55827 - @ramonjd | [Trac ticket](https://core.trac.wordpress.org/ticket/59810) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5622) 
- [x] https://github.com/WordPress/gutenberg/pull/56333 - @jorgefilipecosta | Trac ticket | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5940)

### 📁 plugin

- [x] ~https://github.com/WordPress/gutenberg/pull/57348 - @jorgefilipecosta | Trac ticket | Core backport PR~ 
- [x] ~https://github.com/WordPress/gutenberg/pull/52988 - @adamsilverstein | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~

---
### 📁 block-bindings

> The Block Bindings API will be merged in https://github.com/WordPress/wordpress-develop/pull/5888.

- [x] ~https://github.com/WordPress/gutenberg/pull/57742 | https://github.com/WordPress/gutenberg/pull/57249 - @artemiomorales @SantosGuillamot @michalczaplinski | [Trac ticket](https://core.trac.wordpress.org/ticket/60282) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5888)~
- [x] ~https://github.com/WordPress/gutenberg/pull/58089 - @michalczaplinski | [Trac ticket](https://core.trac.wordpress.org/ticket/60282) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5888)~
- [x] ~https://github.com/WordPress/gutenberg/pull/58105 - @talldan | Trac ticket | Core backport PR~ 
- [x] ~https://github.com/WordPress/gutenberg/pull/58165 - @kevin940726 | Trac ticket | Core backport PR~
- [x] https://github.com/WordPress/gutenberg/pull/58055 - @talldan | [Trac ticket](https://core.trac.wordpress.org/ticket/60282) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5888) 
- [x] https://github.com/WordPress/gutenberg/pull/58587 | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/6072) 

### 📁 interactivity-api

- [x] https://github.com/WordPress/gutenberg/pull/57729 - @c4rl0sbr4v0 | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/57556 - @DAreRodz | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/56302 - @c4rl0sbr4v0 | Trac ticket | Core backport PR 
- [x] ~https://github.com/WordPress/gutenberg/pull/56928 - @dmsnell | Trac ticket | Core backport PR~
- [x] https://github.com/WordPress/gutenberg/pull/56107 - @c4rl0sbr4v0 | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/55404 - @dmsnell | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] https://github.com/WordPress/gutenberg/pull/55739 - @c4rl0sbr4v0 | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/55459 - @DAreRodz | Trac ticket | Core backport PR 
- [x] ~https://github.com/WordPress/gutenberg/pull/55482 - @luisherranz | Trac ticket | Core backport PR~ (part of https://github.com/WordPress/wordpress-develop/pull/5953)
- [x] ~https://github.com/WordPress/gutenberg/pull/55436 - @luisherranz | Trac ticket | Core backport PR~ (part of https://github.com/WordPress/wordpress-develop/pull/5953)
- [x] https://github.com/WordPress/gutenberg/pull/56699 - @luisherranz | [Trac ticket](https://core.trac.wordpress.org/ticket/60232) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5947) 
- [x] ~https://github.com/WordPress/gutenberg/pull/56655 - @luisherranz | Trac ticket | Core backport PR~ (no need to backport)
- [x] https://github.com/WordPress/gutenberg/pull/56143 - @luisherranz | [Trac ticket](https://core.trac.wordpress.org/ticket/56313) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5818) 
### 📁 fonts

> Note: the Fonts Library will be backported in 3 PRs:
> -  [Non-REST API code](https://github.com/WordPress/wordpress-develop/pull/6027) 
> - [REST-API code](https://github.com/WordPress/wordpress-develop/pull/6034)
> - [Register the Google Font collection](https://github.com/WordPress/wordpress-develop/pull/6056)

- [x] https://github.com/WordPress/gutenberg/pull/57735 - @pbking | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/57734 - @pbking | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/55320 - @matiasbenedetto | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] https://github.com/WordPress/gutenberg/pull/54832 - @matiasbenedetto | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] https://github.com/WordPress/gutenberg/pull/57730 - @matiasbenedetto | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/57697 - @jffng | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/54701 - @matiasbenedetto | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/54886 - @pbking | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] ~https://github.com/WordPress/gutenberg/pull/54844 - @pbking | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~
- [x] https://github.com/WordPress/gutenberg/pull/57282 - @matiasbenedetto | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/54795 - @matiasbenedetto | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] https://github.com/WordPress/gutenberg/pull/54748 - @matiasbenedetto | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] https://github.com/WordPress/gutenberg/pull/57364 - @HrithikDalal | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/55393 - @jffng | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/56230 - @jffng | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/56120 - @matiasbenedetto | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/55654 - @Soean | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] https://github.com/WordPress/gutenberg/pull/57569 - @matiasbenedetto | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/55893 - @t-hamano | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/54977 - @matiasbenedetto | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] https://github.com/WordPress/gutenberg/pull/54829 - @matiasbenedetto | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] https://github.com/WordPress/gutenberg/pull/57688 - @creativecoder | Trac ticket | Core backport PR
- [x] https://github.com/WordPress/gutenberg/pull/58193 - @matiasbenedetto | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/58186 - @matiasbenedetto | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/58184 - @matiasbenedetto | Trac ticket | Core backport PR 

### 📁 modules

- [x] https://github.com/WordPress/gutenberg/pull/57437 - @sirreal | [Trac ticket](https://core.trac.wordpress.org/ticket/60233) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5860) 
- [x] https://github.com/WordPress/gutenberg/pull/57256 - @gziolo | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/57396 - @c4rl0sbr4v0 | [Trac ticket](https://core.trac.wordpress.org/ticket/60240) | [Core backport PR 
](https://github.com/WordPress/wordpress-develop/pull/5931)
- [x] https://github.com/WordPress/gutenberg/pull/57231 - @luisherranz | [Trac ticket](https://core.trac.wordpress.org/ticket/56313) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5818) 

### 📁 fonts-api

> the Fonts API has been deprecated and [the files related to it will be removed once the Font Library is merged into Core](https://github.com/WordPress/gutenberg/issues/57959#issuecomment-1900401636)

- [x] ~https://github.com/WordPress/gutenberg/pull/56067 - @arthur791004 | Trac ticket | Core backport PR~
- [x] ~https://github.com/WordPress/gutenberg/pull/55981 - @anton-vlasenko | Trac ticket | Core backport PR~

### 📁 sync

- [x] ~https://github.com/WordPress/gutenberg/pull/55494 - @noahtallen | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~
- [x] ~https://github.com/WordPress/gutenberg/pull/53922 - @jorgefilipecosta | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~
### 📄 blocks.php

- [x] https://github.com/WordPress/gutenberg/pull/57573 - @kevin940726 | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/56235 - @kevin940726 | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/54426 - @getdave | [Trac ticket](https://core.trac.wordpress.org/ticket/59797) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5608) (⚠️ Check for existing WP Core backport)
### 📄 editor-settings.php

- [x] ~https://github.com/WordPress/gutenberg/pull/44214 - @aristath | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~
- [x] ~https://github.com/WordPress/gutenberg/pull/54966 - @ntsekouras | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~
### 📄 class-gutenberg-rest-template-revision-count.php

- [x] ~https://github.com/WordPress/gutenberg/pull/56464 - @tellthemachines | Trac ticket | Core backport PR~ 
### 📄 data-views.php

- [x] ~https://github.com/WordPress/gutenberg/pull/55743 - @jorgefilipecosta | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~
- [x] ~https://github.com/WordPress/gutenberg/pull/55465, https://github.com/WordPress/gutenberg/pull/55918 - @jorgefilipecosta | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~

---
## 📄 client-assets.php

- [x] https://github.com/WordPress/gutenberg/pull/57639 - @youknowriad | Trac ticket | Core backport PR 

---
## 📄 class-wp-theme-json-gutenberg.php

- [x] ~https://github.com/WordPress/gutenberg/pull/56661~ - @ajlende | [Trac ticket](https://core.trac.wordpress.org/ticket/60284) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5896) 
- [x] https://github.com/WordPress/gutenberg/pull/57659 - @tellthemachines | [Trac ticket](https://core.trac.wordpress.org/ticket/60312) | [Core backport PR ](https://github.com/WordPress/wordpress-develop/pull/5917)
- [x] ~https://github.com/WordPress/gutenberg/pull/57165 - @michalczaplinski | Trac ticket | Core backport PR~ 
- [x] https://github.com/WordPress/gutenberg/pull/57005 - @andrewserong | [Trac ticket](https://core.trac.wordpress.org/ticket/60175) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5828)
- [x] https://github.com/WordPress/gutenberg/pull/56793 - @matiasbenedetto | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/56447 - @matiasbenedetto | [Trac ticket](https://core.trac.wordpress.org/ticket/60360) | [Core backport PR ](https://github.com/WordPress/wordpress-develop/pull/5957)
- [x] https://github.com/WordPress/gutenberg/pull/56236 - @tellthemachines | [Trac ticket](https://core.trac.wordpress.org/ticket/60133) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5815) 
- [x] https://github.com/WordPress/gutenberg/pull/55354 - @anton-vlasenko | [Trac ticket](https://core.trac.wordpress.org/ticket/60305) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5911)
- [x] ~https://github.com/WordPress/gutenberg/pull/55313 - @matiasbenedetto | [Trac ticket](url) | Core backport PR (⚠️ Check for existing WP Core backport)~
- [x] https://github.com/WordPress/gutenberg/pull/55261 - @jorgefilipecosta | [Trac ticket](https://core.trac.wordpress.org/ticket/60326) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5934) (⚠️ Check for existing WP Core backport)
- [x] https://github.com/WordPress/gutenberg/pull/53602 - @t-hamano | [Trac ticket](https://core.trac.wordpress.org/ticket/59499) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5427) (⚠️ Check for existing WP Core backport)
- [x] https://github.com/WordPress/gutenberg/pull/58050 - @aaronrobertshaw | [Trac ticket](https://core.trac.wordpress.org/ticket/60343) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5944)
- [x] https://github.com/WordPress/gutenberg/pull/56897 - @andrewserong | [Trac ticket](https://core.trac.wordpress.org/ticket/60365) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5963)
- [x] https://github.com/WordPress/gutenberg/pull/58051 - @aaronrobertshaw | [Trac ticket](https://core.trac.wordpress.org/ticket/60453) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/6053)

---
## 📄 blocks.php

- [x] ~https://github.com/WordPress/gutenberg/pull/55305 - @jorgefilipecosta | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~
- [x] https://github.com/WordPress/gutenberg/pull/54426 - @getdave | [Trac ticket](https://core.trac.wordpress.org/ticket/59797) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5608) (⚠️ Check for existing WP Core backport)
- [x] ~https://github.com/WordPress/gutenberg/pull/44214 - @aristath | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~

---
## 📄 init.php

- [x] ~https://github.com/WordPress/gutenberg/pull/43500 - @mhimon | Trac ticket | Core backport PR~

---
## 📄 global-styles-and-settings.php

- [x] https://github.com/WordPress/gutenberg/pull/57190 - @tellthemachines | [Trac ticket](https://core.trac.wordpress.org/ticket/60134) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5816) 

---
## 📄 class-wp-theme-json-resolver-gutenberg.php

- [x] https://github.com/WordPress/gutenberg/pull/57191 - @tellthemachines | [Trac ticket](https://core.trac.wordpress.org/ticket/60136) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5817) 

---
## 📄 class-wp-duotone-gutenberg.php

- [x] https://github.com/WordPress/gutenberg/pull/54778 - @t-hamano | [Trac ticket](https://core.trac.wordpress.org/ticket/58734) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/4839)
- [x] ~https://github.com/WordPress/gutenberg/pull/55415 - @luisherranz | Trac ticket | Core backport PR (Already backported in WP 6.4)~
- [x] https://github.com/WordPress/gutenberg/pull/54835 - @spacedmonkey | [Trac ticket](https://core.trac.wordpress.org/ticket/59453) | Core backport PR (⚠️ Check for existing WP Core backport)

---
# 📁 phpunit

- [x] https://github.com/WordPress/gutenberg/pull/57866 - @ramonjd | [Trac ticket](https://core.trac.wordpress.org/ticket/60263) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5875) 
- [x] https://github.com/WordPress/gutenberg/pull/57729 - @c4rl0sbr4v0 | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/57735 - @pbking | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/57734 - @pbking | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/57730 - @matiasbenedetto | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/57697 - @jffng | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/57556 - @DAreRodz | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/57569 - @matiasbenedetto | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/54701 - @matiasbenedetto | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/57495 - @andrewserong | [Trac ticket](https://core.trac.wordpress.org/ticket/60175)
- [x] https://github.com/WordPress/gutenberg/pull/57364 - @HrithikDalal | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/56302 - @c4rl0sbr4v0 | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/57282 - @matiasbenedetto | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/57231 - @luisherranz | [Trac ticket](https://core.trac.wordpress.org/ticket/56313) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5818) 
- [x] ~https://github.com/WordPress/gutenberg/pull/56629 - @talldan | Trac ticket | Core backport PR~ 
- [x] https://github.com/WordPress/gutenberg/pull/56143 - @luisherranz | [Trac ticket](https://core.trac.wordpress.org/ticket/56313) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5818) 
- [x] https://github.com/WordPress/gutenberg/pull/55459 - @DAreRodz | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/56447 - @matiasbenedetto | [Trac ticket](https://github.com/WordPress/wordpress-develop/pull/5957) | [Core backport PR ](https://github.com/WordPress/wordpress-develop/pull/5957)
- [x] https://github.com/WordPress/gutenberg/pull/55393 - @jffng | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/56333 - @jorgefilipecosta | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/56528 - @ramonjd | [Trac ticket](https://core.trac.wordpress.org/ticket/59982) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5710) 
- [x] ~https://github.com/WordPress/gutenberg/pull/56464 - @tellthemachines | Trac ticket | Core backport PR~ 
- [x] https://github.com/WordPress/gutenberg/pull/56230 - @jffng | Trac ticket | Core backport PR 
- [x] https://github.com/WordPress/gutenberg/pull/56130 - @tellthemachines | [Trac ticket](https://core.trac.wordpress.org/ticket/60130) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5812)
- [x] https://github.com/WordPress/gutenberg/pull/56107 - @c4rl0sbr4v0 | Trac ticket | Core backport PR 
- [x] ~https://github.com/WordPress/gutenberg/pull/55605 - @scruffian | Trac ticket | Core backport PR~ 
- [x] https://github.com/WordPress/gutenberg/pull/55827 - @ramonjd | [Trac ticket](https://core.trac.wordpress.org/ticket/59810) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5622) 
- [x] https://github.com/WordPress/gutenberg/pull/55739 - @c4rl0sbr4v0 | Trac ticket | Core backport PR 
- [x] ~https://github.com/WordPress/gutenberg/pull/55703 - @dmsnell | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~
- [x] ~https://github.com/WordPress/gutenberg/pull/55482 - @luisherranz | Trac ticket | Core backport PR~ (part of https://github.com/WordPress/wordpress-develop/pull/5953)
- [x] ~https://github.com/WordPress/gutenberg/pull/55436 - @luisherranz | Trac ticket | Core backport PR~ (part of https://github.com/WordPress/wordpress-develop/pull/5953)
- [x] https://github.com/WordPress/gutenberg/pull/55354 - @anton-vlasenko | [Trac ticket](https://core.trac.wordpress.org/ticket/60305) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5911)
- [x] ~https://github.com/WordPress/gutenberg/pull/55313 - @matiasbenedetto | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~
- [x] ~https://github.com/WordPress/gutenberg/pull/44214 - @aristath | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~
- [x] https://github.com/WordPress/gutenberg/pull/54977 - @matiasbenedetto | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] https://github.com/WordPress/gutenberg/pull/54829 - @matiasbenedetto | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] https://github.com/WordPress/gutenberg/pull/53602 - @t-hamano | [Trac ticket](https://core.trac.wordpress.org/ticket/60294) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5902) (⚠️ Check for existing WP Core backport)
- [x] https://github.com/WordPress/gutenberg/pull/54886 - @pbking | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] ~https://github.com/WordPress/gutenberg/pull/54844 - @pbking | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~
- [x] https://github.com/WordPress/gutenberg/pull/54806 - @spacedmonkey | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] https://github.com/WordPress/gutenberg/pull/54489 - @andrewserong | (Already in core)
- [x] https://github.com/WordPress/gutenberg/pull/54651 - @gziolo | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] ~https://github.com/WordPress/gutenberg/pull/56100 - @gaambo | Trac ticket | Core backport PR~
- [x] https://github.com/WordPress/gutenberg/pull/57474 - @andrewserong | [Trac ticket](https://core.trac.wordpress.org/ticket/60175)
- [x] ~https://github.com/WordPress/gutenberg/pull/56719 - @t-hamano | Trac ticket | Core backport PR~ Fixture unit tests are specific to the Gutenberg plugin and should not require backporting to core.
- [x] https://github.com/WordPress/gutenberg/pull/56309 - @aaronrobertshaw | [Trac ticket](https://core.trac.wordpress.org/ticket/60341) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5942) 
- [x] ~https://github.com/WordPress/gutenberg/pull/54801 - @gaambo | Trac ticket | Core backport PR~
- [x] https://github.com/WordPress/gutenberg/pull/55890 - @c4rl0sbr4v0 | Trac ticket | Core backport PR 
- [x] ~https://github.com/WordPress/gutenberg/pull/55542 - @c4rl0sbr4v0 | Trac ticket | Core backport PR~ 
- [x] ~https://github.com/WordPress/gutenberg/pull/55805 - @DAreRodz | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~
- [x] ~https://github.com/WordPress/gutenberg/pull/55714 - @luisherranz | Trac ticket | Core backport PR (Already backported in WP 6.4)~
- [x] ~https://github.com/WordPress/gutenberg/pull/55113 - @aaronrobertshaw | [Trac ticket](https://core.trac.wordpress.org/ticket/59555) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5418) (⚠️ Check for existing WP Core backport)~
- [x] https://github.com/WordPress/gutenberg/pull/54802 - @matiasbenedetto | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] https://github.com/WordPress/gutenberg/pull/54762 - @matiasbenedetto | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] ~https://github.com/WordPress/gutenberg/pull/54836 - @Mamaduka | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)~
- [x] https://github.com/WordPress/gutenberg/pull/54771 - @matiasbenedetto | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)
- [x] https://github.com/WordPress/gutenberg/pull/54763 - @matiasbenedetto | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)

---
# 📁 packages

### 📄 class-wp-style-engine.php

- [x] https://github.com/WordPress/gutenberg/pull/57005 - @andrewserong | [Trac ticket](https://core.trac.wordpress.org/ticket/60175) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5828)
- [x] https://github.com/WordPress/gutenberg/pull/56528 - @ramonjd | [Trac ticket](https://core.trac.wordpress.org/ticket/59982) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5710)

**Comment by gaambo** [comment]
For https://github.com/WordPress/gutenberg/pull/56100 and https://github.com/WordPress/gutenberg/pull/54801 I have already created a Trac ticket but didn't yet get any response if it is necessary. Any help/info is appreciated.
https://core.trac.wordpress.org/ticket/59906

**Comment by t-hamano** [comment]
My understanding is that the code regarding the Font Library (`lib/experimental/fonts`, `phpunit/tests/fonts/font-library`) does not exist in the core at all yet. If we  ship the font library to WP6.5, can it be backported in one PR?

**Comment by andrewserong** [comment]
Thanks for the pings, Dave! 🙇 I've double-checked that we've already done all the background image backports, but I'd missed a small layout backport. I've opened a core PR in https://github.com/WordPress/wordpress-develop/pull/5900 for trac ticket [60292](https://core.trac.wordpress.org/ticket/60292) for that one.

**Comment by Mamaduka** [comment]
Thanks for the ping, Dave. I marked #54836 as completed. It doesn't need the core backport.

**Comment by ntsekouras** [comment]
I marked https://github.com/WordPress/gutenberg/pull/54966 as completed as it doesn't need any backport. PHP changes are just adding an experiment option for the plugin.

**Comment by kt-12** [comment]
#56952 can be marked complete as it doesn't need PHP backport. 6.5 compact files have a shim for backward compatibility and can be removed when WP 6.5 becomes the minimum requirement for this plugin, need not be migrated to the core.

**Comment by mikachan** [comment]
> My understanding is that the code regarding the Font Library (lib/experimental/fonts, phpunit/tests/fonts/font-library) does not exist in the core at all yet. If we ship the font library to WP6.5, can it be backported in one PR?

@t-hamano Thanks for checking this! Yes, the current plan is to backport everything for the Font Library in one PR.

**Comment by mikachan** [comment]
Noting that the Fonts API has been deprecated and the files related to it will be removed once the Font Library is merged into Core. Please see this PR for more info: https://github.com/WordPress/gutenberg/pull/57972

That should mean we can remove/tick the following from the backport list:

- https://github.com/WordPress/gutenberg/pull/56067
- https://github.com/WordPress/gutenberg/pull/55981

**Comment by youknowriad** [comment]
@mikachan when can we expect the font library merge? Can we accelerate there a little bit, even if it means we follow-up with bug fixes?

**Comment by mikachan** [comment]
> when can we expect the font library merge?

We're planning on opening a Core PR as soon as this [feature branch](https://github.com/WordPress/gutenberg/pull/57688) is merged into GB trunk. We're planning on doing this no later than Tuesday 23 Jan, but it's reliant on how quickly we can receive feedback on that branch.

**Comment by youknowriad** [comment]
@mikachan Great, let me know if you need help ensuring people give the feedback on time :)

**Comment by anton-vlasenko** [comment]
Thank you for the ping.

1. https://github.com/WordPress/gutenberg/pull/57030
This one has been backported; I've marked it as such.

2. https://github.com/WordPress/gutenberg/pull/55981

This PR does not require backporting, as per @mikachan's comment here: https://github.com/WordPress/gutenberg/pull/55981#issuecomment-1900393407

3. https://github.com/WordPress/gutenberg/pull/55354
I've created a backport PR for it:

Backport PR: https://github.com/WordPress/wordpress-develop/pull/5911
Trac Ticket: https://core.trac.wordpress.org/ticket/60305

**Comment by artemiomorales** [comment]
I've consolidated the following two issues into one line since they'll be handled by the same backport:
- https://github.com/WordPress/gutenberg/pull/57742
- https://github.com/WordPress/gutenberg/pull/57249

We're currently working on the backport and hope to have it ready soon.

Backport PR: https://github.com/WordPress/wordpress-develop/pull/5888
Trac ticket: https://core.trac.wordpress.org/ticket/60282

cc @michalczaplinski

**Comment by tellthemachines** [comment]
Updated status for the PRs that have already been synced or don't need syncing. (e.g. #57564 was an update _from_ core so doesn't need syncing, #55415 and #55416 went into core during 6.4 Beta and #56464 is a plugin-specific admin task).

**Comment by tellthemachines** [comment]
Note that #56459 needs to be addressed as part of the sync of the `WP_Navigation_Block_Renderer`, whatever shape that takes in core (I'm still inclined to say it should be part of block-library, which would mean it gets synced automatically with the package updates)

**Comment by michalczaplinski** [comment]
> https://github.com/WordPress/gutenberg/pull/57165 - @michalczaplinski | Trac ticket | Core backport PR

It does not need a backport because is was removing backwards-compatibility that was only ever present in GB and never made it to Core. I marked the items as checked in the list.

**Comment by aristath** [comment]
https://github.com/WordPress/gutenberg/pull/44214 - @aristath | Trac ticket | Core backport PR (⚠️ Check for existing WP Core backport)

Should not be backported, these blocks are still experimental and will remain that way for a while :+1:

**Comment by getdave** [comment]
Thanks to everyone who has started on "backporting" their code to WP Core 👏 

Some quick house keeping tasks:

- we would appreciate it if you would only check off your line items once the code is _merged_ into WP Core.
- once the code is safely merged please add the `Backported to WP Core` label to the Gutenberg PR(s) and remove the `Needs PHP Backport` label.
- if you PR does _not_ require a backport please apply ~strikethrough~ formatting (e.g ~strikethrough~) and (if _necessary_) comment on the original PR with a brief explanation as to why it does not require a backport.

Thank you 🙇

**Comment by luisherranz** [comment]
**Script Modules API**

- https://github.com/WordPress/gutenberg/pull/56143
- https://github.com/WordPress/gutenberg/pull/57231

These two belong to the new Script Modules API.

The backport of the Script Modules API started with this PR, which was already committed:

- https://github.com/WordPress/wordpress-develop/pull/5818

@c4rl0sbr4v0 is now working on updating the Gutenberg code with the final version that was committed in WordPress Core, making the necessary adjustments to the blocks, and moving the code to the `compat/wordpress-6.5` folder:

- https://github.com/WordPress/gutenberg/pull/57778

Once that PR is merged, everything should be fine, but we are still waiting until this PR is committed into Core, likely today:

- https://github.com/WordPress/wordpress-develop/pull/5869

---

**Script Modules API: import map polyfill**

- https://github.com/WordPress/gutenberg/pull/56699
- https://github.com/WordPress/gutenberg/pull/56655

These are part of the Script Modules API import map polyfill and no backport exists yet. We'll work on that soon. This is the Trac ticket:

- https://core.trac.wordpress.org/ticket/60232

--- 

**Interactivity API: Server Directive Processing**

- https://github.com/WordPress/gutenberg/pull/55482
- https://github.com/WordPress/gutenberg/pull/55436

These are part of the Server Directive Processing of the Interactivity API. We refactored all the code into a single PR and I will open a PR and a Trac ticket in WordPress Core later today:

- https://github.com/WordPress/gutenberg/pull/58066

---

These three were already included in WP 6.4.

- https://github.com/WordPress/gutenberg/pull/55714
- https://github.com/WordPress/gutenberg/pull/55416
- https://github.com/WordPress/gutenberg/pull/55415

**Comment by dmsnell** [comment]
@getdave on all the commits linked with my name on them, those are taken care of because they originated in Core and were brought back into Gutenberg.

All but the "Fix Directive Processor" commit have the `Backport from WordPress Core` label, which I think I created to try and differentiate those things that _need_ to go into Core from those which are coming _from_ Core.

Let me know if I can adjust what I'm doing to make this clearer next time. Thank you for organizing all this work and for the double- and triple-checks!

**Comment by youknowriad** [comment]
Custom data views and the collaborative editing signaling server are not meant to be released in 6.5 as stable. I've crossed the related tasks above.

**Comment by youknowriad** [comment]
⚠️  **More house keeping**

Some of the backported code live in `lib/experimental` folder or `load.php` or things like that. Please consider moving the backported code into the corresponding `lib/compat/wordpress-6.5` folders instead. This indicates that this code was included in 6.5 and that it can be removed later from the Gutenberg plugin.

**Comment by getdave** [comment]
✅ I updated this PR with the `Backported to Core` label to indicate that the backport has successfully merged into WP Core. I also updated the [PHP Sync Tracking Issue for WP 6.5](https://github.com/WordPress/gutenberg/issues/57959).

**Comment by t-hamano** [comment]
Can someone give me some advice on backporting #57362? This PR makes changes to the following two files.

- `lib/compat/wordpress-6.5/class-wp-navigation-block-renderer.php`
- `packages/block-library/src/navigation/view.js`

The two are closely related, so if we backport just the PHP file, we will likely encounter JavaScript errors on the front end. Do I need to backport this PHP file to coincide with Gutenberg's sync timing?

**Comment by youknowriad** [comment]
@t-hamano Yes, the class being separate from the block is problematic because the block is synced automatically. I think folks are working on this PR to fix that https://github.com/WordPress/gutenberg/pull/57979 (so there's no need to backport manually). Maybe you can help test this PR.

**Comment by tellthemachines** [comment]
Just want to note that I have a backport up for https://github.com/WordPress/gutenberg/pull/57659 but it should only be committed once the first round of package updates goes in, because it depends on a specificity reduction of block-library styles to work properly. The change is non-essential for the package updates so it's fine to commit later.

**Comment by luisherranz** [comment]
We merged yesterday the compat PR for the Script Modules API in Gutenberg:

- https://github.com/WordPress/gutenberg/pull/57778

Everything that is in Core is now in sync and the only thing missing for WP 6.5 on our side is adding a polyfill for browsers that don't support import maps yet, both in Gutenberg and in WordPress Core.

---

We also merged a refactor of the Server Directive Processor (the Interactivity API server code) directly in the 6.5 compat folder in preparation for a WP Core backport:

- https://github.com/WordPress/gutenberg/pull/58066

In this case, I haven't opened the WP Core backport yet, but I'll do it as soon as possible.

**Comment by luisherranz** [comment]
We've just opened the last two backport PRs, so I have updated the description with the links.

- Interactivity API: Server Directive Processing: [Trac](https://core.trac.wordpress.org/ticket/60356) - [PR](https://github.com/WordPress/wordpress-develop/pull/5953)
- Script Modules API: Import map polyfill: [Trac](https://core.trac.wordpress.org/ticket/60348) - [PR](https://github.com/WordPress/wordpress-develop/pull/5947)

**Comment by youknowriad** [comment]
I've committed the initial package update PR, if you had a backport that was waiting for it, you can now go ahead and move forward with it.

It's also a good time to start testing all of your features on WordPress Core with the plugin disabled to check that we didn't miss any backports or anything.

**Comment by t-hamano** [comment]
In #58426, it was found that the navigation view script may not be loaded in WP6.5 alpha. Does this have something to do with backporting PHP?

**Comment by youknowriad** [comment]
@t-hamano I think it's related yes. I'm working with @luisherranz today to try to fix the interactive blocks.

**Comment by youknowriad** [comment]
@t-hamano It should be fixed in https://github.com/WordPress/wordpress-develop/pull/5984

**Comment by luisherranz** [comment]
The [import map polyfill](https://core.trac.wordpress.org/ticket/60356) was committed yesterday. Only one backport left on our side:

- Interactivity API: Server Directive Processing: [Trac](https://core.trac.wordpress.org/ticket/60356) - [PR](https://github.com/WordPress/wordpress-develop/pull/5953)

**Comment by sirreal** [comment]
https://github.com/WordPress/wordpress-develop/pull/5860 is open to backport https://github.com/WordPress/gutenberg/pull/57437 and needs review.

**Comment by michalczaplinski** [comment]
https://github.com/WordPress/wordpress-develop/pull/5888 has been merged so I've marked the Block Bindings-related PRs as "backported".

**Comment by ajlende** [comment]
https://github.com/WordPress/wordpress-develop/pull/5979 is open to backport some theme.json PHPUnit tests and it could use some 👀.

**Comment by aaronrobertshaw** [comment]
https://github.com/WordPress/wordpress-develop/pull/6053 is available to backport a recent fix to block style variations selectors in the theme.json class.

**Comment by talldan** [comment]
I've just opened a new backport PR for pattern overrides - https://github.com/WordPress/wordpress-develop/pull/6055

**Comment by matiasbenedetto** [comment]
The font library has been ported into core  in these 3 PRs:

- https://github.com/WordPress/wordpress-develop/pull/6027
- https://github.com/WordPress/wordpress-develop/pull/6034
- https://github.com/WordPress/wordpress-develop/pull/6056

**Comment by gziolo** [comment]
I landed today https://github.com/WordPress/wordpress-develop/pull/5953 and marked related files as synced for Interactivity API. I also marked other files related to the project that were missed.

**Comment by youknowriad** [comment]
Thanks everyone for your help here. We managed to backported everything on time without too much issues. 

Follow-up work:

 - We invite you all to test your work on WordPress trunk (without the Gutenberg plugin) and ensure that everything is working as expected.
 - We invite you all to check the [6.5 project board](https://github.com/orgs/WordPress/projects/148) and triage/close/punt the issues that you're familiar with and address the bugs ("Todo" column specifically) 
 - From now on, I invite you to add the "Backport to WordPress Beta/RC" label to all of your PRs (mostly bug fixes at this points) that need to be included in WordPress 6.5.

Thank you.
